### DISCUSSION TITLE
Expert help migrating traditional React project to react-static

### DISCUSSION THREAD

**Discussion by augustosamame** [react-static/react-static#1546] [General]
Not really sure if this is the right place to post this, but we're looking for an experienced react-static dev that can help us migrate a traditional React project to react-static. Only some routes will need to be static at first.

We have an Upwork job with more info here:

https://www.upwork.com/jobs/~0172d25674370777b0

Anyone interested please reach out

**Comment by okhobb** [comment]
Curious if you found success. I have a client interested in the same conversion and am trying to teach myself react-static at the moment.

**Comment by augustosamame** [comment]
Nope. Not a peep. We've put this project on hold until we can find a dev that can help us with it.

**Reply by ucimo** [reply]
Any luck?

**Reply by augustosamame** [reply]
Nope. Interested?

**Reply by ucimo** [reply]
No, I also need help from someone who knows react-static...

**Reply by SleeplessByte** [reply]
I think a lot of people are still very affected by COVID (but also using next for example). Overall in the many open source project I contribute to and/or maintain, contributions are much much lower. 

Persist! Keep searching 👯

**Reply by augustosamame** [reply]
IMO react-static is much more focused and therefore a superior solution to NextJS. This one is a paid job so I wonder why it's been so hard to find devs who would be willing to work on it. I guess we'll just have to learn it ourselves. It's just that... deadlines... :(

**Reply by Pixelatex** [reply]
Just because it's a paid job, doesn't mean that it's a well paid job ;) can't see your ad anymore so uncertain.

Many international devs here and even if let's say 10$/hour is a lot in one persons country, in mine, it might not be even worth reading the job description... also, not sure many people actually saw this. I for example only saw this today.

**Reply by augustosamame** [reply]
React-static being sort of a niche skill, I think $30 - $50 / hour would be reasonable. We did mention "expert" in the title so wouldn't expect this to be a $10/hour job. Please reach out if you know someone who would be interested.

![image](https://user-images.githubusercontent.com/16038063/118009719-e99c0980-b313-11eb-9890-97d79b67dccf.png)

**Reply by okhobb** [reply]
I finished the work for my client and I can say that most of the challenge was in refactoring all the references to document/window. So, @augustosamame I don't think you can hope to avoid build-time errors in any easy/automatic way. My advice: before you hire a react-static expert, you should refactor your components to not use window/document directly.

**Reply by ucimo** [reply]
@okhobb Could you please share some more info on this? I can't solve this exact problem in my app. Would you happen to have any videos or blog links to share that deal with it? I'm a newbie in node.js and I don't have much time to learn it right now, although it's on my list.

**Reply by augustosamame** [reply]
@okhobb that's great advice. So we first refactor not to use document / window anywhere, then I reach out to you :) Thanks!

**Reply by augustosamame** [reply]
687 results found in 228 files for `window`
191 results found in 84 files for `document`

this is going to take a while...

**Reply by SleeplessByte** [reply]
That's good advice. If you want to do any type of SSR, you'll almost always have to deal with this. So this is good work even if the end result is not using react-static, but some other tool.

FWIW, the reason I personally don't respond to offers like the one above is that my current freelance rate is 225 USD/hour for things I like. Refactoring someone's project is not something I enjoy so I would apply a multiplier. Just to put things into perspective.
